controllers/nutanixfailuredomainv2_controller_test.go

```diff
@@ -0,0 +1,172 @@
+)
+
+func TestNutanixFailureDomainV2Reconciler(t *testing.T) {
+	g := NewWithT(t)
+
+	_ = Describe("NutanixFailureDomainV2Reconciler", func() {
+		var (
+			ctx         context.Context
+			fdObj       *infrav1.NutanixFailureDomainV2
+			ntnxMachine *infrav1.NutanixMachine
+			reconciler  *NutanixFailureDomainV2Reconciler
+			rstr, rstr2 string
+		)
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			rstr = util.RandomString(10)
+			rstr2 = util.RandomString(12)
+
+			fdObj = &infrav1.NutanixFailureDomainV2{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       infrav1.NutanixFailureDomainV2Kind,
+					APIVersion: infrav1.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fd-test",
+					Namespace: corev1.NamespaceDefault,
+				},
+				Spec: infrav1.NutanixFailureDomainV2Spec{
+					PrismElement: infrav1.PrismElementFailureDomainSpec{
+						PrismElement: infrav1.NutanixResourceIdentifier{
+							Type: infrav1.NutanixIdentifierName,
+							Name: &rstr,
+						},
+						Subnets: []infrav1.NutanixResourceIdentifier{
+							{Type: infrav1.NutanixIdentifierName, Name: &rstr},
+						},
+					},
+				},
+			}
+
+			reconciler = &NutanixFailureDomainV2Reconciler{
+				Client: k8sClient,
+				Scheme: runtime.NewScheme(),
+			}
+
+			ntnxMachine = &infrav1.NutanixMachine{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       infrav1.NutanixMachineKind,
+					APIVersion: infrav1.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "m-test",
+					Namespace: corev1.NamespaceDefault,
+					UID:       utilruntime.NewUUID(),
+				},
+				Spec: infrav1.NutanixMachineSpec{
+					VCPUsPerSocket: 1,
+					VCPUSockets:    2,
+					MemorySize:     resource.MustParse("2Gi"),
+					SystemDiskSize: resource.MustParse("20Gi"),
+					Image: &infrav1.NutanixResourceIdentifier{
+						Type: infrav1.NutanixIdentifierName,
+						Name: &rstr,
+					},
+					Cluster: infrav1.NutanixResourceIdentifier{
+						Type: infrav1.NutanixIdentifierName,
+						Name: &rstr,
+					},
+					Subnets: []infrav1.NutanixResourceIdentifier{
+						{Type: infrav1.NutanixIdentifierUUID, UUID: &rstr},
+					},
+				},
+				Status: infrav1.NutanixMachineStatus{
+					FailureDomain: &fdObj.Name,
+				},
+			}
+		})
+
+		AfterEach(func() {
+			// Delete the failure domain object if exists.
+			_ = k8sClient.Delete(ctx, fdObj)
+
+			// Delete the nutanix machine object if exists.
+			_ = k8sClient.Delete(ctx, ntnxMachine)
+		})
+
+		Context("Update an NutanixFailureDomainV2 Configuration", func() {
+			It("should not allow to update failure domain prism element configuration", func() {
+				// Create the NutanixFailureDomainV2 object and expect creation success
+				g.Expect(k8sClient.Create(ctx, fdObj)).To(Succeed())
+
+				// change the PrismElement configuration
+				fd2 := fdObj.DeepCopy()
+				fd2.Spec.PrismElement.PrismElement.Name = &rstr2
+				err := k8sClient.Update(ctx, fd2)
+				g.Expect(err).To(HaveOccurred())
```

Comment:
lets check the contents of the error to see if they are as expected

Reply:
Added the error message check.